spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -673,11 +673,11 @@ public DataWriter<InternalRow> createWriter(int partitionId, long taskId, long e
       Table table = tableBroadcast.value();
       PartitionSpec spec = table.specs().get(outputSpecId);
       FileIO io = table.io();
-
+      String operationId = queryId + "-" + epochId;
```

Comment:
@amogh-jahagirdar

Are you planning to implement any additional modifications, or does the current fix encompass all scenarios? If the fix is successful, could you provide an estimated time for merging these changes?

Reply:
I still wasn't able to repro this via a unit test so that's why I'm not 100% this really fixes the problem. It's just based on some behaviors that were reported by the original bug reporters.

For ETA, there are no guarantees because it depends on if we as a community conclude if this really is the right fix but I am aiming to get this in the 1.5 release ( maybe the 1.4.3 release since quite a few users have reported this issue and it does lead to a corruption of their data). I'll spend a bit more time digging in on this, and bring this up on the mailing list thread for 1.4.3

Reply:
@amogh-jahagirdar ,

Thank you, We taken source code and added your changes , testing in progress, we will post the update

Reply:
That's great news @VeeraswamyGatta , thanks for testing and verifying! I still have one more idea for trying to repro this via unit test but regardless will move this out of draft.